korangar/src/interface/windows/skill_tree.rs

```diff
@@ -1,27 +1,163 @@
+use std::cmp::Ordering;
+
+use hashbrown::HashMap;
 use korangar_components::skill_box;
+use korangar_interface::element::{Element, ElementBox};
 use korangar_interface::window::{CustomWindow, Window};
-use rust_state::{Path, VecIndexExt};
+use rust_state::{Path, Selector};
 
 use crate::SkillSource;
 use crate::interface::windows::WindowClass;
-use crate::inventory::Skill;
+use crate::inventory::{LearnableSkill, LearnedSkill, LearnedSkillPath};
 use crate::state::localization::LocalizationPathExt;
 use crate::state::theme::InterfaceThemeType;
 use crate::state::{ClientState, ClientStatePathExt, client_state};
 
-pub struct SkillTreeWindow<P> {
-    skills_path: P,
+const SKILL_TREE_COLUMNS: usize = 7;
+
+struct LearnableSkillPath<A> {
+    layout_path: A,
+    index: usize,
+}
+
+impl<A> LearnableSkillPath<A> {
+    fn new(layout_path: A, index: usize) -> Self {
+        Self { layout_path, index }
+    }
+}
+
+impl<A> Copy for LearnableSkillPath<A> where A: Copy {}
+
+impl<A> Clone for LearnableSkillPath<A>
+where
+    A: Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            layout_path: self.layout_path.clone(),
+            index: self.index,
+        }
+    }
+}
+
+impl<A> Path<ClientState, LearnableSkill, false> for LearnableSkillPath<A>
+where
+    A: Path<ClientState, HashMap<usize, LearnableSkill>>,
+{
+    fn follow<'a>(&self, state: &'a ClientState) -> Option<&'a LearnableSkill> {
+        // SAFETY:
+        // Unwrapping is save because of the bounds
```

Comment:
```suggestion
        // Unwrapping is safe because of the bounds
```

(Note: Not really clear *why* to an outside observer/Rust dummy, but that's besides the point)